ddtrace/tracer/tracer.go

```diff
@@ -300,6 +332,11 @@ func (t *tracer) Stop() {
 	case <-t.stopped:
 		return
 	default:
+		if t.statsd != nil {
+			t.statsd.Incr("datadog.tracer.stopped", t.statsTags(), 1)
+			t.statsd.Close()
+			t.statsd = nil
```

Comment:
This is a red flag and a simple programming mistake somewhere can result in a panic.